api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -402,6 +402,10 @@
      */
     Attribute<Boolean> BUFFER_SPOOLING = new Attribute<Boolean>(true);
     /**
+     * Whether to apply back pressure and stall the upstream operator when the downstream operator is slow.
+     */
+    Attribute<Boolean> BUFFER_BACK_PRESSURE = new Attribute<Boolean>(true);
```

Comment:
The spooled buffer isn't durable, it does not survive a container restart and does not contribute to fault-tolerance. The input operator has to be able to replay, either from the source or from its own backup store / WAL. How does publisher blocking or not affect that?